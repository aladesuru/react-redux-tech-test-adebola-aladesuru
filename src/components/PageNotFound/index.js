import React from 'react';
import { Link } from 'react-router-dom';

const PageNotFound = () => {
  return (
    <div>
      <p className="pageNotfound"> Sorry page not found ! </p>
      <Link to="/" className="view-all">
        Back to view all
      </Link>
    </div>
  );
};

export default PageNotFound;
