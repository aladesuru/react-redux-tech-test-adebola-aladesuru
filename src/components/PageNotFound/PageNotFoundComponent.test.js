import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import PageNotFound from './index';
import { BrowserRouter } from 'react-router-dom';

const MockPageNotFound = () => {
  return (
    <BrowserRouter>
      <PageNotFound />
    </BrowserRouter>
  );
};

describe('PageNotFound', () => {
  it('should display Sorry page not found !', () => {
    const wrapper = renderer.create(<MockPageNotFound />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should have a link to go back to home', () => {
    const wrapper = shallow(<PageNotFound />);
    expect(wrapper.find('Link').length).toBe(1);
  });
});
