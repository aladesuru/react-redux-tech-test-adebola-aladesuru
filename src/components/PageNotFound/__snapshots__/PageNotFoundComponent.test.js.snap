// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound should display Sorry page not found ! 1`] = `
<div>
  <p
    className="pageNotfound"
  >
     Sorry page not found ! 
  </p>
  <a
    className="view-all"
    href="/"
    onClick={[Function]}
  >
    Back to view all
  </a>
</div>
`;
